Source/Plugins/Tilemaps/Editor/TilesetEditorModes/AutoTileTilesetEditorMode.cs

```diff
@@ -227,9 +224,14 @@ protected override void OnLayerSelectionChanged(LayerSelectionChangedEventArgs a
 
 			// Update global editor selection, so an Object Inspector can pick up the AutoTile for editing
 			if (selectedNode != null)
+			{
+				this.currentAutoTile = selectedNode.AutoTileInput;
 				DualityEditorApp.Select(this, new ObjectSelection(new object[] { selectedNode.AutoTileInput }));
+			}
 			else
+			{
 				DualityEditorApp.Deselect(this, obj => obj is TilesetAutoTileInput);
+			}
```

Comment:
Why is `this.currentAutoTile` not set to null when the layer is deselected explicitly? In general, how is deselection handled, for example when leaving AutoTile mode? The main concern here is to avoid internally keeping a selection active even though the referenced object has already become invalid.

Cases to check with AutoTile mode open:

- Selecting a different Tileset in the Project View.
- Deleting the selected Tileset in the Project View.
- Removing the selected AutoTile layer from the Tileset.
- Deselecting the selected AutoTile layer explicitly.
- Selecting an AutoTile layer in the Tileset, switching to a different editing mode of the Tileset editor, selecting a different Tileset in the project view, then switching back to AutoTile mode.